src/DiContainer/DiDefinition/DiDefinitionTaggedAs.php

```diff
@@ -202,9 +202,13 @@ private function getKeyFromTagOptionsOrFromKeyDefaultMethod(string $identifier,
                     );
                 }
 
-                if ($taggedAs instanceof DiDefinitionAutowireInterface && str_starts_with($optionKey, 'self::')) {
+                if (!$taggedAs instanceof DiDefinitionAutowireInterface) {
+                    return $optionKey;
+                }
+
+                if (str_starts_with($optionKey, 'self::')) {
```

Comment:
```suggestion
                if (!str_starts_with($optionKey, 'self::')) {
```
остальное ниже перенести